src/util.rs

```diff
@@ -0,0 +1,45 @@
+// Copyright 2019 Developers of the Rand project.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// https://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use core::sync::atomic::{AtomicUsize, Ordering};
+
+// This structure represents a laziliy initialized static usize value. Useful
+// when it is perferable to just rerun initialization instead of locking.
+pub struct LazyUsize(AtomicUsize);
+
+impl LazyUsize {
+    pub const fn new() -> Self {
+        Self(AtomicUsize::new(usize::max_value()))
+    }
+
+    // Runs the init() function at least once, returning the value of some run
+    // of init(). Unlike std::sync::Once, the init() function may be run
+    // multiple times. If init() returns usize::max_value(), the init() function
+    // will always be retried on a future call to unsync_init(). This makes it
+    // ideal for representing failure.
+    pub fn unsync_init(&self, init: impl FnOnce() -> usize) -> usize {
+        // Relaxed ordering is fine, as init() is allowed to run mulitple times.
```

Comment:
Wrong explanation. Probably you know, but the reason is that we only have a single shared variable.

Reply:
https://llvm.org/docs/Atomics.html#atomic-orderings

Reply:
Oh duh, I was overthinking my explanation. Fixed. 

I also renamed the `lazy` module to be `util_libc` as I'm anticipating we will need more `libc` based utility functions.